pkg/referrerstore/oras/oras.go

```diff
@@ -180,6 +187,9 @@ func (store *orasStore) GetBlobContent(ctx context.Context, subjectReference com
 		ref := fmt.Sprintf("%s@%s", subjectReference.Path, digest)
 		resolvedBlobDesc, err = repository.Blobs().Resolve(ctx, ref)
 		if err != nil {
+			if strings.Contains(err.Error(), "401") {
```

Comment:
Is there a error code object we can validate on? strings.contains is risky as the name of registry could also contain numbers? This should also be extracted this into a common method so there is a single method to validate the error codes. Are there also other error code we want to evict, maybe 400, 401, 403 ? Some errors codes can be retried (e.g. 429), some are not retriable like (401).  I think we want to evict the item if its a non retriable error.

Reply:
this makes a lot of sense. I can look into the error more to see if we can use the error type to narrow down if should be retriable.

Reply:
So looks like the ORAS v2 API extracts the errors and formats the error message into specific form here: https://github.com/oras-project/oras-go/blob/49bd44ce628cfedbe10ecfca76727044f0ffdaab/registry/remote/repository.go#L704. I'm not sure how to get around using `strings.Contains`. I updated the `addAuthCache` method to take in a list of error codes (right now only 400, 401, 403). and using the matching identifier `code <error code>` to isolate the error. Still seems a bit clunky. Any other ideas?